pkg/azurefile/azurefile.go

```diff
@@ -858,6 +890,84 @@ func (d *Driver) ResizeFileShare(ctx context.Context, subsID, resourceGroup, acc
 	})
 }
 
+// CopyFileShare copies a fileshare in the same storage account
+func (d *Driver) CopyFileShare(ctx context.Context, req *csi.CreateVolumeRequest, accountKey string, shareOptions *fileclient.ShareOptions, storageEndpointSuffix string, secrets map[string]string) error {
+	if shareOptions.Protocol == storage.EnabledProtocolsNFS {
+		return fmt.Errorf("protocol nfs is not supported for volume cloning")
+	}
+	resourceGroupName, accountName, srcFileShareName, _, _, subsID, err := GetFileShareInfo(req.GetVolumeContentSource().GetVolume().GetVolumeId())
+	dstFileShareName := shareOptions.Name
+	if err != nil {
+		return status.Error(codes.NotFound, err.Error())
+	}
+	klog.V(2).Infof("begin to copy fileshare %s to %s", srcFileShareName, dstFileShareName)
+	if srcFileShareName == "" {
+		return fmt.Errorf("srcFileShareName is empty")
+	}
+	if dstFileShareName == "" {
+		return fmt.Errorf("dstFileShareName is empty")
+	}
+	return wait.ExponentialBackoff(d.cloud.RequestBackoff(), func() (bool, error) {
+		var copyErr error
+		var out []byte
+		if len(secrets) > 0 {
+			accountSasToken, err := getAccountSasToken(secrets, accountName)
+			if err != nil || accountSasToken == "" {
+				klog.V(2).Infof("get account sas token from secrets err: %s, generate sas token for account(%s)", err.Error(), accountName)
+				accountSasToken, err = GenerateSASToken(accountName, accountKey, storageEndpointSuffix)
+				if err != nil {
+					return true, err
+				}
+			}
+			exists, err := d.fileClient.existFileShare(accountName, accountKey, srcFileShareName)
+			if err != nil {
+				return true, err
+			}
+			if !exists {
+				return true, fmt.Errorf("source fileshare(%s) does not exist", srcFileShareName)
+			}
+			srcPath := fmt.Sprintf("https://%s.file.%s/%s%s", accountName, storageEndpointSuffix, srcFileShareName, accountSasToken)
+			dstPath := fmt.Sprintf("https://%s.file.%s/%s%s", accountName, storageEndpointSuffix, dstFileShareName, accountSasToken)
+
+			klog.V(2).Infof("copy fileshare %s to %s", srcFileShareName, dstFileShareName)
+			out, copyErr = exec.Command("azcopy", "copy", srcPath, dstPath, "--recursive", "--check-length=false").CombinedOutput()
```

Comment:
Can you please add a note to docs that it needs azcopy and thus physically copies data? Maybe with a small table about how fast it could be - if I have a volume with 1TB of data (in say 100 MB files), how long will `CreateVolume` take?